tools/readme_linter/assert.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"runtime"
+	"sort"
+
+	"github.com/yuin/goldmark/ast"
+)
+
+//type for all linter assert methods
+type T struct {
+	filename       string
+	markdown       []byte
+	newlineOffsets []int
+
+	fails int
+}
+
+// called by all assert functions that involve a node
+func (t *T) printFailedAssert(n ast.Node, format string, args ...interface{}) {
+	t.printFile(n)
+	fmt.Printf(format+"\n", args...)
+	//t.printRule(3)
```

Comment:
are these for debugging? same question in `assertf`

Reply:
You're talking about the `t.printRule()` calls? They show the line in go code that failed. Linters don't usually do that but since the code is available I thought it might be useful. I was planning to show it to the team and see what you thought.

Reply:
yes, and in practice, that sounds like a really nice usability feature to save a step.